orchagent/nexthopkey.h

```diff
@@ -15,9 +17,24 @@ struct NextHopKey
     string              alias;          // incoming interface alias
     uint32_t            vni;            // Encap VNI overlay nexthop
     MacAddress          mac_address;    // Overlay Nexthop MAC.
+    LabelStack          label_stack;    // MPLS label stack
 
     NextHopKey() = default;
-    NextHopKey(const std::string &ipstr, const std::string &alias) : ip_address(ipstr), alias(alias), vni(0), mac_address() {}
+    NextHopKey(const std::string &str, const std::string &alias) : alias(alias), vni(0), mac_address()
+    {
+        std::size_t label_delimiter = str.find(LABELSTACK_DELIMITER);
```

Comment:
I guess how to encode the action into next hop string needs to be resolved.

Reply:
@smaheshm No, as I stated in another comment thread, there is no need for explicit NH action in FRR/Linux MPLS model.

Reply:
Action has been encoded in the string now.